src/peer_manager.rs

```diff
@@ -118,7 +118,7 @@ pub enum PeerState {
     /// Connected peer is a joining node and waiting for approval of routing.
     Candidate,
     /// We are connected to the peer who is a full node.
-    Node,
+    Node(bool),
```

Comment:
Better add a comment explaining what the `bool` is. Or better, use a named field: `Node { was_joining: bool }` (just a suggestion. Feel free to use any name you think is appropriate).